eslint.config.mjs

```diff
@@ -0,0 +1,42 @@
+// @ts-check
+
+import eslint from '@eslint/js';
+import { defineConfig } from 'eslint/config';
+import tseslint from 'typescript-eslint';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import globals from 'globals';
+
+export default defineConfig(
+  eslint.configs.recommended,
+  tseslint.configs.recommended,
+  eslintConfigPrettier, // Disable formatting eslint rules
+  {
+    ignores: ['dist/', 'node_modules/']
+  },
+  {
+    rules: {
+      '@typescript-eslint/no-explicit-any': 'off', // Allow the usage of `any` in the project
+      "@typescript-eslint/no-unused-vars": ["error", { 
+        // Ignore error for unused args and caught errors if with '_' prefix
+        "argsIgnorePattern": "^_",
+        "caughtErrorsIgnorePattern": "^_",
+      }]
+    },
+  },
+  {
+    "files": ["test/**/*.spec.ts"],
+    "rules": {
+      '@typescript-eslint/no-unused-expressions': 'off', // Allow unused expressions in test files, for compatibility with 'chai'
+      '@typescript-eslint/no-unsafe-function-type': 'off', // Allow unsafe Function type in test files
```

Comment:
We can remove `'@typescript-eslint/no-unsafe-function-type': 'off'` and fix two non-compliant places by using `NextFunction` from `express` instead of `Function` which triggers this violation.